src/main/java/gyro/azure/compute/VirtualMachineResource.java

```diff
@@ -952,6 +1197,38 @@ public void update(GyroUI ui, State state, Resource current, Set<String> changed
             .withDataDiskDefaultStorageAccountType(StorageAccountTypes.fromString(getStorageAccountTypeDataDisk()))
             .withTags(getTags());
 
+        if (changedFieldNames.contains("data-disks")) {
+            Map<String, Integer> currentDataDiskIdsToLun = new HashMap<>();
+            for (Map.Entry<Integer, VirtualMachineDataDisk> dataDiskEntry : virtualMachine.dataDisks().entrySet()) {
+                currentDataDiskIdsToLun.put(dataDiskEntry.getValue().id(), dataDiskEntry.getKey());
+            }
+
+            Set<String> wantedDataDiskIds = getDataDisks()
+                    .stream()
+                    .map(DiskResource::getId)
+                    .filter(s -> client.disks().getById(s) != null)
+                    .collect(Collectors.toSet());
+
+            Set<String> diskIdsToRemove = new LinkedHashSet<>(currentDataDiskIdsToLun.keySet());
+            diskIdsToRemove.removeAll(wantedDataDiskIds);
+            for (String diskIdToRemove : diskIdsToRemove) {
+                if (currentDataDiskIdsToLun.get(diskIdToRemove) != null) {
+                    update.withoutDataDisk(currentDataDiskIdsToLun.get(diskIdToRemove));
+                    ui.write("\n    Removing Data Disk %s", diskIdToRemove);
+                }
+            }
+
+            Set<String> disksIdsToAdd = new LinkedHashSet<>(wantedDataDiskIds);
+            disksIdsToAdd.removeAll(currentDataDiskIdsToLun.keySet());
+            for (String diskIdToAdd : disksIdsToAdd) {
+                Disk diskToAdd = client.disks().getById(diskIdToAdd);
+                if (diskIdToAdd != null) {
+                    update.withExistingDataDisk(diskToAdd);
+                    ui.write("\n    Adding Data Disk %s", diskIdToAdd);
```

Comment:
Seems unnecessary to write output here. The diff should be clear what is going to be added.

Reply:
This was written because the diff only stated that the `data-disk` field was changed, but omitted what changes were made to the field. Is there a better way to let Gyro know the specifics?